libs/lb/lb-rs/src/repo/mod.rs

```diff
@@ -58,23 +58,35 @@ impl<'a> LbTx<'a> {
 
 impl Lb {
     pub async fn ro_tx(&self) -> LbRO<'_> {
+        let start = std::time::Instant::now();
+
         // let guard = tokio::time::timeout(std::time::Duration::from_secs(1), self.db.read())
         //     .await
         //     .unwrap();
 
         let guard = self.db.read().await;
 
+        if start.elapsed() > std::time::Duration::from_millis(100) {
+            tracing::warn!("readonly transaction lock acquisition took {:?}", start.elapsed());
+        }
+
         LbRO { guard }
     }
 
     pub async fn begin_tx(&self) -> LbTx<'_> {
+        let start = std::time::Instant::now();
+
         // let mut guard = tokio::time::timeout(std::time::Duration::from_secs(1), self.db.write())
         //     .await
         //     .unwrap();
 
         let mut guard = self.db.write().await;
         let tx = guard.begin_transaction().unwrap();
 
+        if start.elapsed() > std::time::Duration::from_millis(100) {
+            tracing::warn!("readwrite transaction lock acquisition took {:?}", start.elapsed());
+        }
+
```

Comment:
nice touch

you may want to move this up one line to just time the `write().await`